src/istgt.c

```diff
@@ -2633,6 +2643,38 @@ void *timerfn(void *ptr __attribute__((__unused__)))
 		}
 		while((conn = (CONN *)(istgt_queue_dequeue(&backupconns))) != NULL)
 			istgt_queue_enqueue(&closedconns, conn);
+
+		clock_gettime(clockid, &now);
+		timesdiff(clockid, last_check, now, diff);
+		ms = diff.tv_sec * 1000;
+		ms += diff.tv_nsec / 1000000;
+
+		if (ms > check_interval1) {
+			check_interval1 = check_interval2;
+			MTX_LOCK(&specq_mtx);
+			TAILQ_FOREACH(spec, &spec_q, spec_next) {
+				MTX_LOCK(&spec->complete_queue_mutex);
+				lu_task = (ISTGT_LU_TASK_Ptr)istgt_queue_first(&spec->complete_queue);
+				if (lu_task) {
+					lu_cmd = &lu_task->lu_cmd;
+					clock_gettime(clockid, &now);
+					timesdiff(clockid, lu_cmd->times[0], now, diff);
+					ms = diff.tv_sec * 1000;
+					ms += diff.tv_nsec / 1000000;
+					if (ms > check_interval2) {
+						ISTGT_NOTICELOG("LU:%lu CSN:0x%x TT:%x OP:%2.2x:%x:%s(%lu+%u) not responded since %d seconds\n",
+						    lu_cmd->lun, lu_cmd->CmdSN, lu_cmd->task_tag, lu_cmd->cdb0, lu_cmd->status, lu_cmd->info, lu_cmd->lba, lu_cmd->lblen, ms / 1000);
+					}
+					if (ms < check_interval1)
+						check_interval1 = ms;
```

Comment:
this doesn't seem to be correct. With this, there will be frequent checking if I'm not wrong.
Most of the times, queue will have IOs which are less than 1 sec. So, this check will be happening more than once in a second even though print doesn't come.

Reply:
that's right. We don't need to update `check_interval1` here.